jerry-core/mem/mem-heap.cpp

```diff
@@ -188,7 +188,7 @@ typedef struct
 /**
  * Heap
  */
-mem_heap_t mem_heap;
+mem_heap_t mem_heap JERRY_HEAP_SECTION_ATTR;
```

Comment:
The ```JERRY_HEAP_SECTION_ATTR``` is set to ```__attribute__((section(.jerry_mem)))``` by IoT.js build script. With ```[variable] JERRY_HEAP_SECTION_ATTR``` expression we can set the ```[variable]``` to use section ```.jerry_mem```. The .jerry_mem is defined in nuttx linker script.

With this we can force the jerry-heap to take place in sram.